lib/finch/pool_manager.ex

```diff
@@ -26,26 +34,45 @@ defmodule Finch.PoolManager do
     end
   end
 
-  def lookup_pool(key) do
-    case Registry.lookup(PoolRegistry, key) do
+  def lookup_pool(registry, key) do
+    case Registry.lookup(registry, key) do
       [] ->
         :none
 
       [{pid, _}] ->
         {:ok, pid}
+
+      pids ->
+        # TODO implement alternative strategies
+        {pid, _} = Enum.random(pids)
+        {:ok, pid}
     end
   end
 
-  def start_pool(key) do
-    DynamicSupervisor.start_child(PoolSup, {Finch.Pool, key})
+  def start_pools(registry_name, shp) do
+    {:ok, config} = Registry.meta(registry_name, :config)
+    GenServer.call(config.manager_name, {:start_pools, shp})
   end
 
-  def init(_) do
-    children = [
-      {DynamicSupervisor, name: PoolSup, strategy: :one_for_one},
-      {Registry, [keys: :unique, name: PoolRegistry]},
-    ]
+  def handle_call({:start_pools, shp}, _from, state) do
+    reply = do_start_pools(shp, state)
+    {:reply, reply, state}
+  end
+
+  defp do_start_pools(shp, config) do
+    {count, size} = pool_config(config, shp)
+    pool_args = {shp, config.registry_name, size}
 
-    Supervisor.init(children, strategy: :one_for_one)
+    Enum.map(1..count, fn _ ->
+      DynamicSupervisor.start_child(config.supervisor_name, {Finch.Pool, pool_args})
```

Comment:
```suggestion
      {:ok, pid} = DynamicSupervisor.start_child(config.supervisor_name, {Finch.Pool, pool_args})
      pid
```

So we make sure they have all started properly?